### ISSUE TITLE
HLS video is not resuming after network issues

### ISSUE THREAD

**Issue by nicklive** [videojs/http-streaming#378]
## Description

Hello, I have an issue with video.js+http-streaming, that I can consistently reproduce on basicly all hls streams. Video.js is not resuming live stream after network issues (slow connection and resuming to normal).

Here [reduced test case](http://jsfiddle.net/h1rfL8mk/1/) but for reproducing you need to throttle your connection with chrome devtools (please see _Steps to reproduce_).

## Steps to reproduce

1. Open [link](http://jsfiddle.net/h1rfL8mk/1/) 
2. Wait for loading example and video start
3. Open chrome devtools set throttling for _100 kb/s_ 
4. Wait about 50 seconds (or a bit more)
5. After set throttling to _online_

### Expected
Logically after **Steps to reproduce** video should resume back to live

### Actual
video.js+http-streaming try to load infinity times one segment and aborting it, video is not resuming

### Error output
Error output contains a lot of

```
video.js:142 VIDEOJS: ERROR: DOMException: Failed to set the 'duration' property on 'MediaSource': The 'updating' attribute is true on one or more of this MediaSource's SourceBuffers.
    at HtmlMediaSource.addSeekableRange_ (https://unpkg.com/video.js@7.4.1/dist/video.js:49141:44)
    at addSeekableRange (https://unpkg.com/video.js@7.4.1/dist/video.js:55039:36)
    at PlaylistLoader.<anonymous> (https://unpkg.com/video.js@7.4.1/dist/video.js:55054:15)
    at PlaylistLoader.data.dispatcher (https://unpkg.com/video.js@7.4.1/dist/video.js:1818:33)
    at trigger (https://unpkg.com/video.js@7.4.1/dist/video.js:1954:27)
    at PlaylistLoader.EventTarget.trigger (https://unpkg.com/video.js@7.4.1/dist/video.js:2432:5)
    at PlaylistLoader.haveMetadata (https://unpkg.com/video.js@7.4.1/dist/video.js:39151:14)
    at https://unpkg.com/video.js@7.4.1/dist/video.js:39089:17
    at Object.callback (https://unpkg.com/video.js@7.4.1/dist/video.js:40086:9)
    at cbOnce (https://unpkg.com/video.js@7.4.1/dist/video.js:7220:17)
```

## Additional Information
There are 3 segments in samlpe hls playlist, but I tried 30 segments in playlist and problem repeats but after more delay (2 min) in 4 step.

### videojs-http-streaming version
what version of videojs-http-streaming does this occur with?
videojs-http-streaming 1.6.0

### videojs version
what version of videojs does this occur with?
video.js 7.4.1

### Browsers
what browsers are affected? please include browser and version for each
Chrome 71.0

### Platforms
Windows 10 64-bit

### Other Plugins
none

### Other JavaScript
none

**Comment by lewma** [comment]
I get the exact same behavior.  My videojs session recursively displays:

VIDEOJS: ERROR: DOMException: Failed to set the 'duration' property on 'MediaSource': The 'updating' attribute is true on one or more of this MediaSource's SourceBuffers.
    at HtmlMediaSource.addSeekableRange_ (http://localhost:4300/vendor.js:269446:44)
    at addSeekableRange (http://localhost:4300/vendor.js:275344:36)
    at PlaylistLoader.<anonymous> (http://localhost:4300/vendor.js:275359:15)
    at PlaylistLoader.data.dispatcher (http://localhost:4300/vendor.js:221309:33)
    at trigger (http://localhost:4300/vendor.js:221445:27)
    at PlaylistLoader.EventTarget.trigger (http://localhost:4300/vendor.js:221923:5)
    at PlaylistLoader.haveMetadata (http://localhost:4300/vendor.js:259445:14)
    at http://localhost:4300/vendor.js:259375:17
    at Object.callback (http://localhost:4300/vendor.js:260390:9)
    at cbOnce (http://localhost:4300/vendor.js:226932:17)

**Comment by thecotne** [comment]
i have same issue 

when connection speed drops player can't recover after connection is recovered

**Comment by alessandro-bottamedi** [comment]
Same issue here, to reproduce simply set "Slow 3G" in network Tap in google Chrome DevTools... after the connection is restored the problem persist. 
Someone have found a temporary solution?

**Comment by coolicer** [comment]
I got this problem too. how to fix it?

**Comment by danrossi** [comment]
I'm trying to find a solution to try and play if a hls live stream isn't available. Once it goes to an error , you can't click to play again. Any ideas ?

**Comment by danrossi** [comment]
This causes a stack issue, as the player tries to autoload instead of after clicking. Can't turn off autoloading

```
player.on("error", () => {

        	player.pause();
        	player.trigger("ended");
        
        	player.reset();
        
        	player.src(player.currentSrc());

        });
```

**Comment by coolicer** [comment]
@danrossi It may not catch on error event

**Comment by alessandro-bottamedi** [comment]
The error must be catched at the line 50294 (ver 7.5.1):

`this.nativeMediaSource_.duration = end;`

I modified it in this way:

```
  try {
    this.nativeMediaSource_.duration = end;
  } catch (e) {
    if (typeof streamCustomError === "function") { 
      streamCustomError(e);
    }
    return this.trigger('error');
  } 
```

Where `streamCustomError(e)` is my custom function to handle this specific error....

However I noticed that after about 5 minutes the video streaming starts again correctly.

**Comment by danrossi** [comment]
An offline stream will go to error but no way to click to try and play again. Setting the src will try and start loading again.

**Comment by danrossi** [comment]
An example fiddle I found. With a non existent manifest, you cant click to try and load again

https://jsfiddle.net/m3sf8nav/

**Comment by coolicer** [comment]
@alessandro-bottamedi What did you do in `streamCustomError ` function

**Comment by alessandro-bottamedi** [comment]
> @alessandro-bottamedi What did you do in `streamCustomError ` function

I reset and reload the video streaming after a timeout...

**Comment by coolicer** [comment]
@alessandro-bottamedi Thank you, I'll try it later.

**Comment by json-m** [comment]
this could wind up being a billing issue for those using cdns to deliver hls. i discovered this bug watching metrics in my own cdn account after flipping the switch in my player for users that are very distant from my hls relay server to help with buffering.

using a single client to test (2.5Mbps~ video), set browser throttle speed to about 200kbps. after a while it requests the same ts chunk hundreds of times (300+ if i just let it go forever)

![](https://cdn.discordapp.com/attachments/333279437204488193/575015748997480460/unknown.png)

balloons up to quite a bit of bandwidth (17x !!)

![](https://cdn.discordapp.com/attachments/333279437204488193/575016337441816605/unknown.png)

eventually the bandwidth dips off at the end there as it starts abandoning requests. the player stops abandoning requests after a bit which just starts this cycle over again.

![](https://cdn.discordapp.com/attachments/333279437204488193/575016494497398786/unknown.png)

since the player is still trying to dl all of these segments it is requesting, the player is presumably making the flakey connectivity problem severely worse for the end user since they are now trying to download the same segment hundreds of times.

i tested this on the latest firefox and chrome at the time of writing and you can see the videojs version in the screenshot (7.5.4).

**Comment by movAX13h** [comment]
When the connection drops completely, videojs tries all lower quality settings and keeps the lowest. 
When the connection is re-established, it resumes playback at lowest quality and does not return to a better quality stream. Is there a way to circumvent this behaviour? Especially in cases where videojs is used in applications that are not controlled by any user and complete reloads should be avoided, this is critical.

**Comment by pacepace** [comment]
Firefox is doing this as well. 

Videojs version
video.js 7.4.3 and 7.5.5

Browsers
Firefox 68.0b7

Platforms
Windows 10 64-bit

**Comment by vksbansal** [comment]
any update?

**Comment by arjitsachdeva** [comment]
I'm facing the same issue as outlined above - tested in the latest version of http-streaming (v1.11.1) and also on the latest version of video.js (v7.6.5)

**Comment by snowdream** [comment]
any update?

**Comment by bruinxs** [comment]
I got this problem too. Any update?

**Comment by danrossi** [comment]
I've built an offline reconnect feature. Tested with HLS and Wowza. 

https://videojs.electroteque.org/offline

**Comment by lusase** [comment]
Can I prevent this error through plug-ins or other methods?

**Comment by flexmodule** [comment]
> @alessandro-bottamedi Thank you, I'll try it later.
Have you solved it？

**Comment by flexmodule** [comment]
any update?

**Comment by dkirilenko** [comment]
any update? I got the same issue

**Comment by picheli20** [comment]
I've notice that this bug was introduced on version `7.2.1`, downgrading to version `7.2.0` solves this problem (at least till someone comes up with more stable solution)

**Comment by thecotne** [comment]
in [videojs@7.2.1](https://github.com/videojs/video.js/releases/tag/v7.2.1) `videojs/http-streaming` has been updated from `1.2.1` to `1.2.3` here is merge request https://github.com/videojs/video.js/pull/5368/files

i have checked latest videojs `videojs@7.7.5` with `videojs/http-streaming@1.2.1` it works 

and `videojs@7.7.5` with `videojs/http-streaming@1.2.2` has this bug

so regression is somewhere here https://github.com/videojs/http-streaming/compare/v1.2.1...v1.2.2

there is total of 3 commits that can possibly cause any issue (others are tests or docs modifications)

- https://github.com/videojs/http-streaming/commit/6c68761b5d06d951d7ed20cbd5bb63f324ad2c47
- https://github.com/videojs/http-streaming/commit/7c683354ec21e60fbd4d90f8a82e37586fef9f87
- https://github.com/videojs/http-streaming/commit/960ee797e1665b5176f66490897c2f548f88f4e0

from this commits one is `mux.js` update i have checked old `mux.js` version and it has same bug

from remaining two one is typeofs minification related (less likely to cause any problem)

and second is updates of seeking behavior and i think this commit is a problem https://github.com/videojs/http-streaming/commit/6c68761b5d06d951d7ed20cbd5bb63f324ad2c47 (committed by @gesinger and merged by @forbesjo)

**Comment by gkatsev** [comment]
Thanks for the deeper dive, it was helpful. We're looking into this and should have a better answer on this some time today.

**Comment by gesinger** [comment]
Thanks to everyone for narrowing it down. We were actually just looking into this, and narrowed down the issue to the same commits pointed out by @thecotne, as per @picheli20 's finding of what version first started the issue.

The commit which causes the issue is, as @thecotne identified, https://github.com/videojs/http-streaming/pull/161 , a change where we updated how we perform seeks in VHS.

We looked a little further into it, and found that this change may have uncovered a deeper issue in VHS related to duration updates.

When the network connection is throttled for live streams, a couple of things happen:

1) Manifest refreshes continue, and are generally successful due to the small number of bytes downloaded
2) Segment requests continue, but often fail due to the larger number of bytes required and a slow connection

Since the manifest refreshes continue to happen, the seekable range (live window) continues to move forwards, while, due to the segment request failures, the current time stays behind, and eventually, the current time is before the start of the seekable range. In this case, where we "fall off" the live window, we seek to the end of the seekable range (this is performed by playback wacher).

The seek goes through the middleware, and tries to update the tech's current time (as per the PR noted). We placed a `debugger` in the `seekTo` function, and noted that the `currentTime` on the native video element does not update. Instead of seeking to the end of the seekable range, the native video element's current time remains unchanged.

Checking back in the console log (and as noted in the issue), we saw that we tried to update the duration of the native media source at one point, but the duration update failed because the media source was in the updating state (it was busy). This happened because we don't queue the duration change as we should, and do for other operations in source updater.

This chain of events leads to a loop of the playback watcher attempting to correct the current time to the end of the seekable range and the media element adjusting the seek position back to the old duration value which hasn't been updated.

So the source of the issue may be the duration update not being queued, and the PR brought with it a case that surfaced the issue. But it's also possible that the seeks themselves pose a problem. We're currently investigating both.

On a side note, we are in the process of working on the `master` branch of VHS (rather than the `1.x` branch), which includes many changes to the segment loading process (primarily, transmuxes happen before appends, rather than during appends). The branch does not present such problems. Duration changes on the native media source are queued there, so that leads us to the belief that the root of the problem lies there. We don't have an expected release yet of the switchover to the `master` branch, but this is provided mainly as evidence of our theory, and if anyone wants to test the behavior there, we are happy to hear your own results.

We'll continue our investigation and updates. Thanks again for everyone helping provide samples and findings!

**Comment by gkatsev** [comment]
Thanks @gesinger!

If anyone here has a spare moment to test their streams against the master branch and report back issues (preferably in a separate issue) that would be highly appreciated.

**Comment by gkatsev** [comment]
So, @gesinger looked into it some more, and we believe that making the changes to the duration updates like we have in the master branch right now is too risky and would take too long to be worth it.
We're looking into the implications of reverting 6c68761 in the 1.x branch (which is currently in Video.js) as that will be the simplest fix and will allow us to focus on getting the `master` branch ready for usage. We'll keep this issue updated with what we decide and should have a decision by next week.

I'd once again like to ask for help in testing the master branch. The more eyes on it the better!

**Comment by ArMikael** [comment]
Hi guys, do you have any updates or solution regarding this issue?

**Comment by gkatsev** [comment]
We believe that the solution right now is to revert the middleware change. We have PR #777 open but it needs a few tweaks because we'll merge it. It will likely be ready this week.

**Comment by jacekhajduk** [comment]
video.js@7.7.6 this bug is still:
`video.es.js:97 VIDEOJS: ERROR: DOMException: Failed to set the 'duration' property on 'MediaSource': The 'updating' attribute is true on one or more of this MediaSource's SourceBuffers.`

**Comment by gkatsev** [comment]
The PR for the fix hasn't been merged and isn't available in Video.js yet.

**Comment by nvuhung** [comment]
Hello, any update for this issue?

**Comment by DispatchCommit** [comment]
@gkatsev Testing with 1.13.3 which I believe includes the merged middleware related PR linked above and still experiencing issues. I haven't been able to fully analyze the cause, but for the first ~10-30 seconds after an HLS livestream is published, if I attempt to load the stream I encounter this error. Haven't fully investigated the circumstances yet though to identify a root issue.

```
D:\bitwave>npm ls @videojs/http-streaming
bitwave.tv@1.14.1 D:\bitwave
`-- video.js@npm:@bitwave/video.js@7.8.4
  `-- @videojs/http-streaming@1.13.3
```

Attempting to causes this error:
```
VIDEOJS: TypeError: Failed to set the 'currentTime' property on 'HTMLMediaElement': The provided double value is non-finite.
    at Html5.setCurrentTime (video.es.js?f0e2:19314)
    at set (video.es.js?f0e2:9563)
    at Player.eval (video.es.js?f0e2:22964)
    at Player.ready (video.es.js?f0e2:3932)
    at Player.techCall_ (video.es.js?f0e2:22962)
    at Player.currentTime (video.es.js?f0e2:23226)
    at SeekBar.handleMouseMove (video.es.js?f0e2:13239)
    at SeekBar.handleMouseDown (video.es.js?f0e2:12275)
    at SeekBar.handleMouseDown (video.es.js?f0e2:13183)
    at HTMLDivElement.data.dispatcher (video.es.js?f0e2:1899) Video is not ready. (Video.js)
```

and eventually:
```
video.es.js?f0e2:97 VIDEOJS: ERROR: DOMException: Failed to set the 'duration' property on 'MediaSource': The 'updating' attribute is true on one or more of this MediaSource's SourceBuffers.
    at HtmlMediaSource.addSeekableRange_ (webpack-internal:///./node_modules/video.js/dist/video.es.js:36594:42)
    at addSeekableRange (webpack-internal:///./node_modules/video.js/dist/video.es.js:42983:34)
    at PlaylistLoader.eval (webpack-internal:///./node_modules/video.js/dist/video.es.js:42998:13)
    at PlaylistLoader.data.dispatcher (webpack-internal:///./node_modules/video.js/dist/video.es.js:1939:31)
    at trigger (webpack-internal:///./node_modules/video.js/dist/video.es.js:2075:25)
    at PlaylistLoader.EventTarget.trigger (webpack-internal:///./node_modules/video.js/dist/video.es.js:2478:3)
    at PlaylistLoader.haveMetadata (webpack-internal:///./node_modules/video.js/dist/video.es.js:27487:12)
    at eval (webpack-internal:///./node_modules/video.js/dist/video.es.js:27414:15)
    at Object.eval [as callback] (webpack-internal:///./node_modules/video.js/dist/video.es.js:28472:7)
    at cbOnce (webpack-internal:///./node_modules/@videojs/xhr/index.js:95:21)
```

As well as an `hls-video-underflow` event: `hls-video-underflow: {type: "usage", name: "hls-video-underflow", target: video#streamplayer_html5_api.vjs-tech, ...`

Once I get a chance to debug this further I'll open a new issue with any findings.

**Comment by DispatchCommit** [comment]
Reverting to http-streaming 1.12.2 for video.js's dependancy resolves my issue, though the `VIDEOJS: ERROR: DOMException: Failed to set the 'duration' property on 'MediaSource': The 'updating' attribute is true on one or more of this MediaSource's SourceBuffers.` error still appears in the console.

It's definitely related to attempting to load a stream that hasn't fully populated it's manifest though. Refreshing the page once the manifest has all the segments required for it's DVR functionality it runs fine, but attempting to load the stream prior to the manifest being fully populated results in video.js re-requesting .ts segments at an increasing rate until it stalls out it's available bandwidth.

**Comment by gkatsev** [comment]
Have a sample of how your manifest looks like during that transitionary period? Also, we have a mux.js update pending in the 1.x branch with some fixes that we need to make.

Also, can you try it out on https://videojs-http-streaming.netlify.app/? It's what we hope to release really soon, it's a big update with lots of fixes, changes, and refactors.

**Comment by DispatchCommit** [comment]
> sample of how your manifest looks like during that transitionary period

During transition the playlist will appear like this:

```
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-MEDIA-SEQUENCE:0
#EXT-X-TARGETDURATION:2
#EXT-X-DISCONTINUITY
#EXTINF:2.000,
1588854648283.ts
```

```
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-MEDIA-SEQUENCE:0
#EXT-X-TARGETDURATION:2
#EXT-X-DISCONTINUITY
#EXTINF:2.000,
1588854648283.ts
#EXTINF:2.000,
1588854650282.ts
```
(keeps adding additional .ts segments until DVR is filled)
Then (this is when issues begin):
```
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-MEDIA-SEQUENCE:1
#EXT-X-TARGETDURATION:2
#EXTINF:2.000,
1588854650282.ts
#EXTINF:2.000,
1588854652300.ts
#EXTINF:2.000,
1588854654297.ts
#EXTINF:2.000,
1588854656294.ts
#EXTINF:2.000,
1588854658291.ts
#EXTINF:2.000,
1588854660288.ts
#EXTINF:2.000,
1588854662285.ts
#EXTINF:2.000,
1588854664281.ts
#EXTINF:2.000,
1588854666303.ts
#EXTINF:2.000,
1588854668299.ts
#EXTINF:2.000,
1588854670295.ts
#EXTINF:2.000,
1588854672293.ts
#EXTINF:2.000,
1588854674291.ts
#EXTINF:2.000,
1588854676312.ts
#EXTINF:2.000,
1588854678312.ts
#EXTINF:2.000,
1588854680312.ts
#EXTINF:2.000,
1588854682312.ts
#EXTINF:2.000,
1588854684297.ts
#EXTINF:2.000,
1588854686312.ts
#EXTINF:2.000,
1588854688312.ts
#EXTINF:2.000,
1588854690312.ts
#EXTINF:2.000,
1588854692312.ts
#EXTINF:2.000,
1588854694312.ts
#EXTINF:2.000,
1588854696312.ts
#EXTINF:2.000,
1588854698301.ts
#EXTINF:2.000,
1588854700296.ts
#EXTINF:2.000,
1588854702292.ts
#EXTINF:2.000,
1588854704289.ts
#EXTINF:2.000,
1588854706286.ts
#EXTINF:2.000,
1588854708283.ts
```
Note: `#EXT-X-MEDIA-SEQUENCE:0` now increases.
Note: `#EXT-X-DISCONTINUITY` no longer appears (unless we have dropped frames).

---
> can you try it out on https://videojs-http-streaming.netlify.app/

Seems to run well there.

**Comment by DispatchCommit** [comment]
@gkatsev Also of note is that when the manifest file is first initially populating, the player does not show the live indicator, and it does not have a progress bar. And it does not show up afterwards either, I have to reload the player again to get the live indicator and progress bar to appear.

**Comment by tvkit** [comment]
> Also, can you try it out on https://videojs-http-streaming.netlify.app/? It's what we hope to release really soon, it's a big update with lots of fixes, changes, and refactors.

@gkatsev  the DOMException is no longer occurring in our environment and playback is much more reliable. Occasionally getting network errors that we're hoping to mitigate via the player's error handler.

**Comment by hassanabidpk** [comment]
Hi, Is this merged and available in official release of `video.js`? 
I tried `video.js@7.8.2` but this problem still persists.

**Comment by DispatchCommit** [comment]
I saw a comment somewhere that this issue may have originated after `7.2.0`. I'm on `2.0.0-rc2` and still having this issue, may try to downgrade and compare.

**Comment by DispatchCommit** [comment]
Confirmed still an issue using all latest versions of http-streaming and video.js. Also reproducing on https://videojs-http-streaming.netlify.app/
Streams loaded before manifest have fully populated will cause errors, and live UI will not function. 

https://github.com/videojs/http-streaming/issues/378#issuecomment-597717070 seems spot on for my experience.

**Comment by Raino** [comment]
Any news?

**Comment by gkatsev** [comment]
The latest VHS 1.x release line should've fixed the issue as far as we know.
The same fix was made in the main branch after 2.0.0-rc.2 was tagged and should be in the 2.0.0 pre-release tag.

In our testing, this should be fixed. Please try latest VHS versions and let us know if you are still experiencing the issue.

**Comment by ronneylira** [comment]
@gkatsev when using react and npm how do I make sure I'm on the latest version of http-streaming?
as I understood the video.js package already contains http-streaming.

what I am trying to do is:

```
import videojs from 'video.js';
import '@videojs/http-streaming';
```

is there a way to make sure I'm on the correct version?

**Comment by DispatchCommit** [comment]
@ronneylira I use a custom video.js build and then publish that to NPM and in my package.json:
```
"video.js": "npm:@bitwave/video.js@^7.8.5",
```
to replace video.js with a custom build. 

I'm working on updating my build to the latest RC to test.

**Comment by gkatsev** [comment]
If you install `@videojs/http-streaming@2.0.0` locally and import it as above, you should get it.
Though, the latest Video.js (7.8.4) should be using VHS 1.13.2 which includes the fix in the 1.x release line. So, theoretically, you don't need to include a custom VHS version.

**Comment by DispatchCommit** [comment]
I deployed `7.8.4` to our dev site for testing, and this issue still occurs. I've set the DVR on the playlist to be 5 minutes, and it seems to require about 30 seconds of DVR before the player recognizes the DVR is available. 

I'll try updating to `2.0.0` and including like above and see if that behaves any better.

**Comment by DispatchCommit** [comment]
updated to 7.8.4 and http-streaming 2.0.0 and imported as shown above.

My server waits 15 seconds before sending the notice to clients that a streamer has gone live, and when that occurs, the player fully reloads (resets) and updates the source for the player. I increased that delay to 30 seconds and it still exhibits the same behavior. I'd hate to have to further increase the delay to 1 minute or greater, but even then I'm not positive the DVR would show initially. If I wait 30 seconds after the player reloads, and then trigger a second player, the live UI and DVR show up normally, but I cannot seem to get the DVR and Live UI to appear on the first load / reset.

![image](https://user-images.githubusercontent.com/45262335/87075094-aee9b200-c1d4-11ea-9085-ff0961549bfd.png)

**Comment by DispatchCommit** [comment]
Once _more_ than 30 seconds of DVR are available, it appears to load fine though. I can try reloading after 35 seconds to confirm this, but it would seem fairly consistent that 30 seconds is the required DVR time before things work properly. 

![image](https://user-images.githubusercontent.com/45262335/87076062-148a6e00-c1d6-11ea-8b2d-732e81677495.png)

**Comment by Raino** [comment]
@DispatchCommit My friend, could you please share your player style file? :D

**Comment by DispatchCommit** [comment]
@Raino of course! :) 
Here's a link to the sass file in our frontend repo: 
https://github.com/bitwave-tv/bitwave/blob/dev/assets/style/stream-player.scss
perma-link (in case we change the style at a later date): 
https://github.com/bitwave-tv/bitwave/blob/947b0645c407fcf10565d57aba03b73ed07375f6/assets/style/stream-player.scss

**Comment by DispatchCommit** [comment]
I THINK I FIGURED IT OUT!

https://github.com/videojs/video.js/pull/6409
> Use the old liveui for live videos where seekableEnd - seekableStart is less than 30s

```js
const defaults = {
  // Number of seconds of live window (seekableEnd - seekableStart) that
  // a video needs to have before the liveui will be shown.
  trackingThreshold: 30
};
```

This explains why I'm not seeing the live UI when my stream has less than 30 seconds of data!

Adjusting the threshold value to a more optimistic 5 seconds should ensure we avoid this issue. I need to reduce my server delays back down to 10 seconds to test and confirm this is the cause.

**Comment by asoulliereHT** [comment]
I'm experiencing what seems to be this issue.  rolling back to 7.2.0 as mentioned above appears to resolve this.  any updates on this issue?

**Comment by danrossi** [comment]
A client just alerted me to my offline feature. Videojs is also now failing to internally retry on client side disconnection. And the interval is not consistent. Either first retry or second retry. It needs a health check on the first retry. I let it retry to a configured interval before starting the offline reconnect sequence, but in the case of client side disconnection, no error event is sent and it fails to retry !  I got work arounds working.

**Comment by renepardon** [comment]
> I've notice that this bug was introduced on version `7.2.1`, downgrading to version `7.2.0` solves this problem (at least till someone comes up with more stable solution)

I tried to go back to 7.2.0. Didn't worked for me.

**Comment by asoulliereHT** [comment]
> > I've notice that this bug was introduced on version `7.2.1`, downgrading to version `7.2.0` solves this problem (at least till someone comes up with more stable solution)
> 
> I tried to go back to 7.2.0. Didn't worked for me.

i actually ended up upgrading to the latest and just found that my implementation was a lot of the problem

**Comment by gkatsev** [comment]
I guess I never circled back here. We've been working on a new ABR algorithm that is based on buffer length, as part of that we also switched to a timer for doing ABR related things. I believe that this will greatly improve it. Please try it out by enabling `experimentalBufferBasedABR` on VHS.

**Comment by shyamvala** [comment]
Hi,

Just curious if there is any resolution to this. I am on the 7.9.7 vjs version and 2.8.0 of http-streaming. I am seeing the playback stop after a segment fails to load (in dev console, i am seeing a cancelled request, i am not seeing any dom errors). The playback just stops after and I have waited for over 20mins to just see if it ever comes back but not really. 

VJS: 7.9.7
HttpStreaming: 2.8.0
Browser: Opera
`                       
controls: true,
autoplay: true,
html5: {
vhs: {
	overrideNative: true,
	enableLowInitialPlaylist: true,
	experimentalBufferBasedABR: true
}
nativeAudioTracks: false,
nativeVideoTracks: false
}
`

Wondering if its a case of just using an older version of the libraries or some additional configuration to turn on.

**Comment by mayeaux** [comment]
> I THINK I FIGURED IT OUT!
> 
> [videojs/video.js#6409](https://github.com/videojs/video.js/pull/6409)
> 
> > Use the old liveui for live videos where seekableEnd - seekableStart is less than 30s
> 
> ```js
> const defaults = {
>   // Number of seconds of live window (seekableEnd - seekableStart) that
>   // a video needs to have before the liveui will be shown.
>   trackingThreshold: 30
> };
> ```
> 
> This explains why I'm not seeing the live UI when my stream has less than 30 seconds of data!
> 
> Adjusting the threshold value to a more optimistic 5 seconds should ensure we avoid this issue:
> 
> ```js
> const player = videojs('vid', {
>   liveTracker: {
>     trackingThreshold: 5
>   }
> });
> ```
> 
> I need to reduce my server delays back down to 10 seconds to test and confirm this is the cause.
> 
> Edit: Setting the threshold to 0 resolved this issue for me!
> 
> ![image](https://user-images.githubusercontent.com/45262335/87113990-57733280-c224-11ea-8d74-ad5c40625340.png)
> 
> I'm happy with the behavior now, _and_ I can reduce all my timers to just delay by 15 seconds and have no loading issues. What a pain in the butt that was to fully identify and resolve.

Thanks buddy works for me, well appreciated I would have spent a whole day to figure that out lol
